### ISSUE TITLE
end_segment should not break when context_missing is LOG_ERROR

### ISSUE THREAD

**Issue by jaysonsantos** [aws/aws-xray-sdk-python#98]
Right now, calls to end_segment will break when `context_missing='LOG_ERROR'`, should that happen?

```python
>>> xray_recorder.end_segment()
cannot find the current segment/subsegment, please make sure you have a segment open
No segment to end
cannot find the current segment/subsegment, please make sure you have a segment open
Traceback (most recent call last):
...
  File ".../aws-xray-sdk/aws_xray_sdk/core/recorder.py", line 229, in end_segment
    if self.current_segment().ready_to_send():
AttributeError: 'NoneType' object has no attribute 'ready_to_send'
```

**Comment by haotianw465** [comment]
Thank you for reporting this issue. This is a bug and we will work on a fix soon.

**Comment by haotianw465** [comment]
The issue is fixed in 2.2.0 https://pypi.org/project/aws-xray-sdk/. Closing.
